### PR TITLE
Remove NOTREACHED_IN_MIGRATION usage in decentralized_dns

### PR DESCRIPTION
**Pull request by yrliou**
And some random places in the code base.

Resolves https://github.com/brave/brave-browser/issues/42224

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/ai_chat/renderer/ai_chat_resource_sniffer_throttle_unittest.cc [hunk 1]
```diff
@@ -183,7 +182,7 @@ class MockDelegate : public blink::URLLoaderThrottle::Delegate {
       case MOJO_RESULT_SHOULD_WAIT:
         return 0;
       default:
-        NOTREACHED_IN_MIGRATION();
+        NOTREACHED();
     }
     return 0;
```
Comment (cdesouza-chromium) [inline components/ai_chat/renderer/ai_chat_resource_sniffer_throttle_unittest.cc:187]: Is it possible to remove the `default:` entry and just have `NOTREACHED()` outside the switch? That would help detecting new cases, if they got added.

File: components/decentralized_dns/content/decentralized_dns_opt_in_page.cc [hunk 2]
```diff
@@ -63,7 +63,7 @@ void DecentralizedDnsOptInPage::CommandReceived(const std::string& command) {
       controller()->Proceed();
       break;
     default:
-      NOTREACHED_IN_MIGRATION() << "Unsupported command: " << command;
+      NOTREACHED() << "Unsupported command: " << command;
   }
```
Comment (cdesouza-chromium) [inline components/decentralized_dns/content/decentralized_dns_opt_in_page.cc:67]: Ditto

File: components/decentralized_dns/content/ens_offchain_lookup_opt_in_page.cc [hunk 3]
```diff
@@ -61,7 +61,7 @@ void EnsOffchainLookupOptInPage::CommandReceived(const std::string& command) {
       controller()->Proceed();
       break;
     default:
-      NOTREACHED_IN_MIGRATION() << "Unsupported command: " << command;
+      NOTREACHED() << "Unsupported command: " << command;
   }
```
Comment (cdesouza-chromium) [inline components/decentralized_dns/content/ens_offchain_lookup_opt_in_page.cc:65]: Ditto

File: components/ai_chat/renderer/ai_chat_resource_sniffer_throttle_unittest.cc (hunk 1 above)
Comment (yrliou) [reply components/ai_chat/renderer/ai_chat_resource_sniffer_throttle_unittest.cc]: Force pushed an update to fix this, https://github.com/brave/brave-core/pull/26494/commits/babdb91402f60ba31f30a258259bef6919149be9

File: components/decentralized_dns/content/decentralized_dns_opt_in_page.cc (hunk 2 above)
Comment (yrliou) [reply components/decentralized_dns/content/decentralized_dns_opt_in_page.cc]: Force pushed an update to fix this, https://github.com/brave/brave-core/commit/babdb91402f60ba31f30a258259bef6919149be9

File: components/decentralized_dns/content/ens_offchain_lookup_opt_in_page.cc (hunk 3 above)
Comment (yrliou) [reply components/decentralized_dns/content/ens_offchain_lookup_opt_in_page.cc]: Force pushed an update to fix this, https://github.com/brave/brave-core/commit/babdb91402f60ba31f30a258259bef6919149be9

**Review by cdesouza-chromium — APPROVED** [review]
LGTM

File: components/decentralized_dns/content/decentralized_dns_interstitial_controller_client.cc [hunk 4]
```diff
@@ -63,8 +63,7 @@ void DecentralizedDnsInterstitialControllerClient::SetResolveMethodAndReload(
   } else if (IsSnsTLD(request_url_.host_piece())) {
     pref_name = kSnsResolveMethod;
   } else {
-    NOTREACHED_IN_MIGRATION();
-    return;
+    NOTREACHED();
```
Comment (iefremov) [inline components/decentralized_dns/content/decentralized_dns_interstitial_controller_client.cc:66]: where does `request_url_` comes from? It seems risky to have notreached (crash) if it's not really clear how it is formed

File: components/decentralized_dns/content/decentralized_dns_opt_in_page.cc [hunk 5]
```diff
@@ -130,7 +130,7 @@ void DecentralizedDnsOptInPage::PopulateInterstitialStrings(
                                IDS_SNS_OPT_IN_PRIMARY_PARAGRAPH),
                            {sol_domain, sns_wiki_link}, nullptr));
   } else {
-    NOTREACHED_IN_MIGRATION();
+    NOTREACHED();
```
Comment (iefremov) [inline components/decentralized_dns/content/decentralized_dns_opt_in_page.cc:133]: same question about the request_url

File: components/decentralized_dns/content/decentralized_dns_interstitial_controller_client.cc (hunk 4 above)
Comment (yrliou) [reply components/decentralized_dns/content/decentralized_dns_interstitial_controller_client.cc:66]: https://github.com/brave/brave-core/blob/master/components/decentralized_dns/content/decentralized_dns_navigation_throttle.cc#L57-L68
https://github.com/brave/brave-core/blob/master/components/decentralized_dns/content/decentralized_dns_navigation_throttle.cc#L87-L96
So I think it's safe.

File: components/decentralized_dns/content/decentralized_dns_opt_in_page.cc (hunk 5 above)
Comment (yrliou) [reply components/decentralized_dns/content/decentralized_dns_opt_in_page.cc:133]: See above.

**Review by iefremov — APPROVED** [review]

**Pull request merged**